qtox/build_qt_macos.sh

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+# SPDX-License-Identifier: GPL-3.0-or-later AND MIT
+# Copyright (c) 2024 The TokTok team
+
+set -euo pipefail
+
+readonly SCRIPT_DIR="$(dirname "$(realpath "$0")")"
+
+source "$SCRIPT_DIR/build_utils.sh"
+
+parse_arch --dep "qt" --supported "macos-arm64 macos-x86_64" "$@"
+
+QT_VERSION="6.8.1"
+
+brew install qt@6
```

Comment:
Why is this needed?

Don't we build our own Qt6 from scratch here, why do we also need to install Qt6 from a package manager before we build ours?

Reply:
Yes, Qt needs a Qt to bootstrap itself. After that, we don't need it to build additional modules.

Reply:
Looks like I'm wrong. It's only needed for cross-compiling.

Reply:
Yep, it's needed only for cross-compiling, and it's preferred if you build a native Qt off the same source (use native Qt of the same version) that you are cross-compiling, instead of using whatever Qt version is in the package manager. First you build a minimal native QtBase off the source, then you cross-compile a foreign QtBase with `-qt-host-path /native/qt/build/install/prefix` parameter.